### ISSUE TITLE
YouTube page reloads entirely instead of reloading only video frame, after clicking a video in the playlist.

### ISSUE THREAD

**Issue by smartfonreddit** [sniklaus/youtube-watchmarker#8]
Sign in. Have some videos in the Watch Later playlist. Click on one of them. The video loads and on the right-top corner there is list of upcoming videos in the playlist. Click on another video.

WHAT HAPPENS: The whole YouTube page reloads and takes a long time to appear.

WHAT SHOULD HAPPEN: All elements of the page should stay intact and only the video frame/description should reload with the new video data.

The general performance of YouTube seems to be heavily impacted by the extension. 

Using Windows 10 64bit, Firefox 60 Beta.

**Comment by sniklaus** [comment]
Thank you for bringing this up and for your detailed description. I am afraid that reloading the page is intentional due to the way the new YouTube design has been implemented. The extension needs a way to determine when the content of the webpage changed to examine its (updated) content and accordingly update the database of watched videos.

Unfortunately, the new YouTube design reuses previously utilized elements for different content which makes this rather difficult. I also tried to directly hook into YouTube's client-side code, but it is quite a behemoth. I therefore opted to make each link force-load the new content as a new page, allowing the extension to only monitor changes during a page load.

I have noticed that the new YouTube design is quite slow in general, with or without the extension. I am  personally still using the old YouTube design due to this reason.
